Tests/Common/Orders/Fills/ImmediateFillModelTests.cs

```diff
@@ -349,7 +349,91 @@ public void PerformsStopMarketFillSell()
+            security.SetMarketPrice(new IndicatorDataPoint(Symbols.SPY, Noon, 101.4m));
+
+            var fill = model.Fill(new FillModelParameters(
+                security,
+                order,
+                new MockSubscriptionDataConfigProvider(config),
+                Time.OneHour)).OrderEvent;
+
+            Assert.AreEqual(0, fill.FillQuantity);
+            Assert.AreEqual(0, fill.FillPrice);
+            Assert.AreEqual(OrderStatus.None, fill.Status);
+
+            security.SetMarketPrice(new IndicatorDataPoint(Symbols.SPY, Noon, 99.99m));
+
+            fill = model.Fill(new FillModelParameters(
+                security,
+                order,
+                new MockSubscriptionDataConfigProvider(config),
+                Time.OneHour)).OrderEvent;
+
+            // this fills worst case scenario, so it's min of asset/stop price
+            Assert.AreEqual(order.Quantity, fill.FillQuantity);
+            Assert.AreEqual(Math.Max(security.Price, order.LimitPrice), fill.FillPrice);
+            Assert.AreEqual(OrderStatus.Filled, fill.Status);
+        }
 
+        [Test]
+        public void PerformsLimitIfTouchedFillSell()
+        {
+            var model = new ImmediateFillModel();
+            var order = new LimitIfTouchedOrder(Symbols.SPY, -100, 101.5m, 102m, Noon);
+            var config = CreateTradeBarConfig(Symbols.SPY);
+            var security = new Security(
+                SecurityExchangeHoursTests.CreateUsEquitySecurityExchangeHours(),
+                config,
+                new Cash(Currencies.USD, 0, 1m),
+                SymbolProperties.GetDefault(Currencies.USD),
+                ErrorCurrencyConverter.Instance,
+                RegisteredSecurityDataTypesProvider.Null,
+                new SecurityCache()
+            );
+            security.SetLocalTimeKeeper(TimeKeeper.GetLocalTimeKeeper(TimeZones.NewYork));
+            security.SetMarketPrice(new IndicatorDataPoint(Symbols.SPY, Noon, 101.51m));
+
+            var fill = model.Fill(new FillModelParameters(
+                security,
+                order,
+                new MockSubscriptionDataConfigProvider(config),
+                Time.OneHour)).OrderEvent;
+
+            Assert.AreEqual(0, fill.FillQuantity);
+            Assert.AreEqual(0, fill.FillPrice);
+            Assert.AreEqual(OrderStatus.None, fill.Status);
+
+            security.SetMarketPrice(new IndicatorDataPoint(Symbols.SPY, Noon, 102.1m));
+
+            fill = model.Fill(new FillModelParameters(
+                security,
+                order,
+                new MockSubscriptionDataConfigProvider(config),
+                Time.OneHour)).OrderEvent;
+
+            // this fills worst case scenario, so it's min of asset/stop price
+            Assert.AreEqual(order.Quantity, fill.FillQuantity);
+            Assert.AreEqual(Math.Min(security.Price, order.LimitPrice), fill.FillPrice);
```

Comment:
`FillModel` doesn't at worst case scenario anymore? Same for `PerformsLimitIfTouchedFillBuy`